File: ConsoleApp1/ConsoleApp1/DauSachKhoaHoc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
    class DauSachKhoaHoc:DauSach
    {
        protected string sChuyenNganh;
		protected List<QuyenSachKhoaHoc> lDSQuyenSach;
		public string ChuyenNganh
		{
			get { return this.sChuyenNganh; }
		}
		public List<QuyenSachKhoaHoc> DSQuyenSach
		{
			get { return this.lDSQuyenSach; }
			set { this.lDSQuyenSach = value; }
		}
		public DauSachKhoaHoc()
		{
			XacDinhThue();
			this.lDSQuyenSach = new List<QuyenSachKhoaHoc>();
		}
        public DauSachKhoaHoc(string MaDauSach, string Tua, string NXB, string TacGia, double Gia, string ChuyenNganh) : base(MaDauSach, Tua, NXB, TacGia, Gia)
        {
            this.sChuyenNganh = ChuyenNganh;
			XacDinhThue();
			this.lDSQuyenSach = new List<QuyenSachKhoaHoc>();
        }
		public override void Nhap()
		{
			base.Nhap();
			Console.Write(" Chuyen nganh: ");
			this.sChuyenNganh = Console.ReadLine();
		}
		public override void Xuat()
		{
			base.Xuat();
			Console.WriteLine(" Chuyen nganh: " + this.sChuyenNganh);
		}
		public override void TinhThanhTien()
		{
			this.dThanhTien = this.dGia + (this.dGia * this.dThue/100);
		}
		public override void XacDinhThue()
		{
			this.dThue= 5;
		}
	}
}

File: Quan_Ly_Thu_Vien/src/DauSachKhoaHoc.java
import java.util.ArrayList;
import java.util.Scanner;

public class DauSachKhoaHoc extends DauSach
{
	protected String sChuyenNganh;
	protected ArrayList <QuyenSachKhoaHoc> lDSQuyenSach;
	
	public String getsChuyenNganh() 
	{
		return sChuyenNganh;
	}

	public void setsChuyenNganh(String sChuyenNganh)
	{
		this.sChuyenNganh = sChuyenNganh;
	}

	public ArrayList<QuyenSachKhoaHoc> getlDSQuyenSach() 
	{
		return lDSQuyenSach;
	}

	public void setlDSQuyenSach(ArrayList<QuyenSachKhoaHoc> lDSQuyenSach)
	{
		this.lDSQuyenSach = lDSQuyenSach;
	}
	public DauSachKhoaHoc()
	{
		XacDinhThue();
		this.lDSQuyenSach = new ArrayList<QuyenSachKhoaHoc>();
	}
	public DauSachKhoaHoc(String MaSach,String Tua,String NhaXuatBan, String TacGia, double DonGia,String ChuyenNganh)
	{
		super(MaSach,Tua, NhaXuatBan,TacGia, DonGia);
		this.sChuyenNganh = ChuyenNganh;
		XacDinhThue();
		this.lDSQuyenSach = new ArrayList<QuyenSachKhoaHoc>();
	}
	@SuppressWarnings("resource")
	@Override
	public void Nhap()
	{
		super.Nhap();
		Scanner sc = new Scanner(System.in);		
		System.out.print("Nhap chuyen nganh : ");
		this.sChuyenNganh = sc.nextLine();
	}	
	@Override
	public void Xuat()
	{
		super.Xuat();
		System.out.println(" Chuyen nganh: " + this.sChuyenNganh);
	}
	@Override 
	public void TinhThanhTien()
	{
	}
	@Override
	public void XacDinhThue()
	{
		this.dThue= 5;
	}
}
